src/HttpClientInterception/DelegateHelpers.cs

```diff
@@ -200,14 +200,18 @@ internal static class DelegateHelpers
             }
         }
 
-        if (predicates.Length == 0)
+        if (predicates.Count == 0)
         {
             throw new InvalidOperationException("At least one predicate must be provided.");
         }
 
-        if (predicates.Length == 1)
+        if (predicates.Count == 1)
         {
-            return message => Task.FromResult(predicates[0](message));
+            return message =>
+            {
+                Predicate<HttpRequestMessage>? predicate = predicates.First();
+                return Task.FromResult(predicate(message));
+            };
```

Comment:
Can just capture it once and re-use rather than grab it from the collection on every invocation.

```suggestion
            var predicate = predicates.First();
            return message => return Task.FromResult(predicate(message));
```